### PR TITLE
endpoint added to retrieve comments via article id - TASK 9

### PR DESCRIPTION
**Pull request by philjacks**
- TASK 9
- endpoint added to retrieve comments via article id
- errors 404 & 400 tested
- comments controllers & models files added
- refactored error handlers into seperate controller file

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -217,7 +217,7 @@ describe("GET /api/articles", () => {
         expect(articles).toBeInstanceOf(Array);
         expect(articles).toHaveLength(12);
         articles.forEach((article) => {
-          expect(article).toMatchObject({
+          expect(article).toStrictEqual({
```
Comment (emilyb93) [inline __tests__/endpoints.test.js:220]: Why did you refactor to this?

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -239,3 +239,42 @@ describe("GET /api/articles", () => {
...
+describe("GET /api/articles/:article_id/comments", () => {
+  test("Status 200 - should return an array of comments via article id with the correct properties", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body: { comments } }) => {
+        expect(comments).toBeInstanceOf(Array);
+        expect(comments).toHaveLength(11);
+        comments.forEach((comment) => {
+          expect(comment).toEqual({
```
Comment (emilyb93) [inline __tests__/endpoints.test.js:252]: we should expect that each comment has the article_id that we are requesting also

File: models/commentsModels.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        created_at,
+        author,
+        body
+        FROM comments
+        WHERE article_id = $1
+    `;
+
+  return db.query(queryStr, [id]).then((data) => {
+    const comments = data.rows;
+    if (comments.length === 0) {
```
Comment (emilyb93) [inline models/commentsModels.js:17]: What if the article exists but there are no comments? Is this still an error?

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Some changes to think about

File: __tests__/endpoints.test.js (hunk 1 above)
Comment (philjacks) [reply __tests__/endpoints.test.js:220]: I was told to make sure the body wasn't being sent in the response & toMatchObject still passed the test, therefore I used toStrictEqual :)

File: __tests__/endpoints.test.js (hunk 2 above)
Comment (philjacks) [reply __tests__/endpoints.test.js:252]: Aplogies. The task doesn't state to include it so just assumed it didn't want it

File: __tests__/endpoints.test.js (hunk 1 above)
Comment (philjacks) [reply __tests__/endpoints.test.js]: sorted

File: models/commentsModels.js (hunk 3 above)
Comment (philjacks) [reply models/commentsModels.js]: sorted :)

**Pull request merged**